File: src/main/java/com/example/warehouseflowmanager/common/api/ApiPaths.java
package com.example.warehouseflowmanager.common.api;

public final class ApiPaths {

    public static final String API_V1 = "/api/v1";

    public static final String HEALTH = API_V1 + "/health";
    public static final String PRODUCTS = API_V1 + "/products";
    public static final String STOCK_MOVEMENTS = API_V1 + "/stock-movements";
    public static final String STORAGE_LOCATIONS = API_V1 + "/storage-locations";

    private ApiPaths() {
    }
}

File: frontend/src/app/core/api/api-paths.ts
export const API_BASE_PATH = '/api/v1';

export const ApiPaths = {
  health: `${API_BASE_PATH}/health`,
  products: `${API_BASE_PATH}/products`,
  lowStockProducts: `${API_BASE_PATH}/products/low-stock`,
  replenishmentCandidates: `${API_BASE_PATH}/products/replenishment-candidates`,
  stockMovements: `${API_BASE_PATH}/stock-movements`,
  stockMovementSummary: `${API_BASE_PATH}/stock-movements/summary`,
  storageLocations: `${API_BASE_PATH}/storage-locations`,
} as const;

export const storageLocationStockOverviewPath = (storageLocationId: number): string =>
  `${ApiPaths.storageLocations}/${storageLocationId}/stock-overview`;
